### ISSUE TITLE
false positives

### ISSUE THREAD

**Issue by pranav318** [aosokin/os2d#7]
Dear Aosokin,

Thank you for the great job here, especially the detailed descriptions how to set-up the experiments and reproduce the results.
how to reduce the false positives in this case
![69](https://user-images.githubusercontent.com/32036977/96417637-2fc23c00-120f-11eb-8aef-61fd011df9f3.png)
the class image is 
![70](https://user-images.githubusercontent.com/32036977/96417706-45cffc80-120f-11eb-82cf-10b12622f2ed.png)

Thankyou

**Comment by aosokin** [comment]
Hi, it looks like in this particular example it's a matter of the threshold for detections. If you set it to 0.59 all the false positives will be gone.
Best,
Anton

**Comment by pranav318** [comment]
yeah but the threshold cannot be generalized  for all cases.

**Comment by aosokin** [comment]
Well, selecting the correct threshold is indeed tricky. There is a trade-off between false positives and undetected objects. The existence of very similar classes makes the problem even harder. If you have data with a lot of similar classes fine-tuning on it can potentially improve performance. We trained the model on the best dataset available to us but there is clear room for improvement.

**Comment by pranav318** [comment]
we have a case were we need to distinguish similar images as above, do you have any recommended method/approach for distinguishing between false and actual class (provided there is not much training data), is there a way for us to take advantage of the color differences in these similar images

**Comment by aosokin** [comment]
We noticed that ImageNet-pretrained weights directly (before fine-tuning) are a bit more sensitive to colors. You can try the OS2D V2-init model which uses them directly. Alternatively, you can merge similar classes into one class with several class images and train an additional classifier to distinguish those - this can be possible without a lot of data.
I'm afraid I do not have significantly more to suggest.

**Comment by pranav318** [comment]
thanks for the help.
also i had a small doubt that do the training of the model require only 1 image per class or the annotated aisle images as in the grozi data set

**Comment by aosokin** [comment]
To train the models, for each class, you need 1 template image and several annotated objects of this class (positives). Optionally, you can use several template images (e.g. use annotated positives as templates) - randomly select one in each batch.

**Comment by pranav318** [comment]
Thank you.
